packages/contracts/contracts/maci/Tally.sol

```diff
@@ -0,0 +1,321 @@
+    }
+
+    super.addTallyResults(
+      voteOptionIndices,
+      results,
+      tallyResultProofs,
+      tallyResultSalt,
+      spentVoiceCreditsHashes,
+      perVOSpentVoiceCreditsHashes
+    );
+
+    if (recipientCount < totalTallyResults) {
+      revert TooManyResults();
+    }
+  }
+
+  /// @inheritdoc TallyBase
+  function addTallyResult(
+    uint256 voteOptionIndex,
+    uint256 tallyResult,
+    uint256[][] calldata tallyResultProof,
+    uint256 tallyResultSalt,
+    uint256 spentVoiceCreditsHash,
+    uint256 perVOSpentVoiceCreditsHash,
+    uint8 voteOptionTreeDepth
+  ) internal override {
+    super.addTallyResult(
+      voteOptionIndex,
+      tallyResult,
+      tallyResultProof,
+      tallyResultSalt,
+      spentVoiceCreditsHash,
+      perVOSpentVoiceCreditsHash,
+      voteOptionTreeDepth
+    );
+
+    totalVotesSquares += tallyResult ** 2;
+  }
+
+  /// @inheritdoc IPayoutStrategy
+  function claim(
+    IPayoutStrategy.Claim calldata params
+  ) public override isInitialized whenNotPaused afterVotingDeadline {
+    uint256 amount = getAllocatedAmount(params.voiceCreditsPerOption, params.totalSpent, params.tallyResult);
+    totalAmount -= amount;
+
+    bool isValid = verifyTallyResult(
+      params.index,
+      params.tallyResult,
+      params.tallyResultProof,
+      params.tallyResultSalt,
+      params.voteOptionTreeDepth,
+      params.spentVoiceCreditsHash,
+      params.perVOSpentVoiceCreditsHash
+    );
+
+    if (!isValid) {
+      revert InvalidTallyVotesProof();
+    }
+
+    IRecipientRegistry.Recipient memory recipient = registry.getRecipient(params.index);
+
+    token.safeTransfer(recipient.recipient, amount);
+  }
+
+  /// @notice Get allocated token amount (without verification).
+  /// @param voiceCreditsPerOption The voice credit options received for recipient
+  /// @param totalSpent The total amount of voice credits spent
+  /// @param tallyResult The tally result for recipient
+  function getAllocatedAmount(
+    uint256 voiceCreditsPerOption,
+    uint256 totalSpent,
+    uint256 tallyResult
+  ) internal view returns (uint256) {
+    uint256 alpha = calculateAlpha(totalSpent);
+    uint256 quadratic = alpha * voiceCreditFactor * tallyResult * tallyResult;
+    uint256 totalSpentCredits = voiceCreditFactor * voiceCreditsPerOption;
+    uint256 linearPrecision = ALPHA_PRECISION * totalSpentCredits;
+    uint256 linearAlpha = alpha * totalSpentCredits;
+
+    return ((quadratic + linearPrecision) - linearAlpha) / ALPHA_PRECISION;
+  }
+
+  /// @notice Calculate the alpha for the capital constrained quadratic formula
+  /// @dev page 17 of https://arxiv.org/pdf/1809.06421.pdf
+  /// @param totalSpent The total amount of voice credits spent
+  function calculateAlpha(uint256 totalSpent) internal view returns (uint256) {
+    uint256 budget = token.balanceOf(address(this));
+
+    uint256 contributions = totalSpent * voiceCreditFactor;
+
+    if (budget < contributions) {
```

Comment:
won't this revert after one claim or is it supposed to be called only once?

Reply:
No, it won't revert. I have tests where all the recipients get the funds and there are several claims.

Reply:
It reverts only if there is not enough token balance of the tally contract